## Домашнє завдання 6: Статистичні тести

0. Зчитайте дані з `data.csv` та запишіть значення колонки data у форматі np.ndarray в змінну data. Ми будемо далі працювати з цією змінною.

In [12]:
!pip install statsmodels

     ---------------------------------------- 0.0/60.8 kB ? eta -:--:--
     ------------------------- ------------ 41.0/60.8 kB 960.0 kB/s eta 0:00:01
     -------------------------------- ----- 51.2/60.8 kB 518.5 kB/s eta 0:00:01
     -------------------------------------- 60.8/60.8 kB 537.6 kB/s eta 0:00:00
   ---------------------------------------- 0.0/9.8 MB ? eta -:--:--
   ---------------------------------------- 0.1/9.8 MB 5.5 MB/s eta 0:00:02
    --------------------------------------- 0.1/9.8 MB 1.7 MB/s eta 0:00:06
    --------------------------------------- 0.2/9.8 MB 2.3 MB/s eta 0:00:05
   - -------------------------------------- 0.4/9.8 MB 2.1 MB/s eta 0:00:05
   - -------------------------------------- 0.5/9.8 MB 2.5 MB/s eta 0:00:04
   --- ------------------------------------ 0.9/9.8 MB 3.5 MB/s eta 0:00:03
   ---- ----------------------------------- 1.0/9.8 MB 3.7 MB/s eta 0:00:03
   -------- ------------------------------- 2.0/9.8 MB 5.6 MB/s eta 0:00:02
   --------


[notice] A new release of pip is available: 24.1.2 -> 24.2
[notice] To update, run: python.exe -m pip install --upgrade pip


In [13]:
import pandas as pd 
import numpy as np

In [14]:
df = pd.read_csv('../data_gen/data.csv')  
data = np.array(df['data'])

Запустіть код нижче. Для коректної робити всіх подальших методів дані мають бути у вигляді одновимірного вектора, тобто мати розмір (100,).

In [15]:
data.shape

(100,)

1. Зчитані дані - це сума покупок в доларах наших покупців на сайті протягом липня 2022 року.

До цього місяця, ми мали продажі в середньому на чек $20. Чи вийшло у нас статистично значущо **підвищити** середній чек за останній місяць?

Оберіть статистичний тест (z-test чи t-test). Виведіть p-значення статистичного тесту та зробіть висновок, чи ми маємо зараз статистично значущу різницю у розмірі середнього чеку.

**Підказка**: тут тест буде одновибірковий.

In [16]:
from statsmodels.stats.weightstats import ztest

In [24]:
ztest_score, p_value = ztest(data, value =20, alternative = 'larger')

In [27]:
ztest_score, p_value

(np.float64(-0.9953477389335053), np.float64(0.8402164165464012))

In [28]:
alpha = 0.05

if(p_value <  alpha):
    print("Відхилити Н0.")
else:
    print("Н0 не може бути відхилена.")

Н0 не може бути відхилена.


P-value > 0.05, тому ми не можему відкинути нульову гіпотезу , а також ztest_score  -0.995 нижче від нульової гіпотези,тому ми не маємо статистично значущу різницю у розмірі середнього чеку.

2. Пройшов час, і ми зібрали дані про продажі за серпень 2022 року, які зберігли у файл `data_2.csv`. Продажів було дещо більше, але чи змінився стат. значущо середній чек порівняно з липнем 2022? Перевірте це використовуючи ztest.

  Виведіть p-значення статистичного тесту та зробіть висновок, чи ми маємо зараз статистично значущу різницю у розмірі середнього чека між двома місяцями.

  **Підказка**: треба провести двовибірковий ztest з Н0 - що середні значення в липні та серпні 2022 року - однакові. Для кращого розуміння ситуації, можливо вам буде цікаво переглянути, які середні показники чеку були в кожен з місяців. Порівняйте свій висновок з простого спостереження середніх значень і висновок статистичного тесту. Чи вони збігаються?

In [29]:
df_2 = pd.read_csv('../data_gen/data_2.csv')  
data_2 = np.array(df_2['data'])

In [30]:
data_2.shape

(120,)

In [32]:
np.mean(data), np.mean(data_2)

(np.float64(19.37755319670441), np.float64(19.37322276512519))

In [34]:
ztest(data, data_2, alternative='two-sided')

(np.float64(0.005246673456808504), np.float64(0.9958137794593356))

P-value (0..995) > 0.05 ,що означає, що ми не можемо відхилити нульову гіпотезу, а також значення Z-score test (0.005) дуже невелике,що вказує на невелику різницю між середніми значеннями липня та серпня.   
З простої перевірки середнії значень двух вибірок ми також побавили,що ці значення -  майже однакові - 19.377 та 19.373.  
Отже, висновок статистичного тесту та висновок зі спостереження середніх значень двох вибірок - 